src/app/pages/visitsCalendar/visitsCalendar.component.ts

```diff
@@ -66,77 +71,92 @@ export class VisitsCalendarComponent implements OnInit {
 
 
     private loadAllPatients() {
-        this.patientService.getPatients().subscribe( (patients) => { this.patients = patients; });
+        this.patientService.getPatients().subscribe((patients) => {
+            this.patients = patients;
+        });
     }
 
-    reloadEvents(){
-
-        // this.http.get(`${this.config.apiUrl}/doctors/${this.id}/slots`)
-        //     .map((response: Response) => response.json())
-        //     .subscribe(slots => {this.events = slots});
-
-        this.events =  [{
-            slotId:"3",
-            title: 'Visit',
-            color: this.colors.yellow,
-            start: new Date()
-        }, {
-            slotId:"4",
-            title: 'Visit',
-            color: this.colors.blue,
-            start: new Date()
-        }];
+    reloadEvents() {
+        //zawsze zaciągamy miesiąc - nawet jak patrzymy na tydzień/dzień (tak prościej :P)
+        var viewStart = new Date(this.viewDate.getFullYear(), this.viewDate.getMonth(), 1);
+        var viewEnd = new Date(this.viewDate.getFullYear(), (this.viewDate.getMonth() + 1) % 12, 1);
+
+        this.timeSlotService.getTimeSlots(this.doctor, viewStart, viewEnd)
+            .subscribe(slots => {
+                var events = slots.map(slot => {
+                    return {
+                        slotId: slot.id,
+                        title: "Visit",
+                        start: new Date(slot.startDateTime),
+                        end: new Date(slot.endDateTime),
+                        color: this.colors.blue
+                    }
+                })
+                this.events = events
+
+                //mamy terminy, teraz sprawdzamy które są zajęte!
+                for (var i = 0; i < events.length; i++) {
+                    (function (event: VisitEvent, self: VisitsCalendarComponent) { // w tej funkcji nie mam dostępu do `this`
```

Comment:
Wydaje mi się, że nie musisz robić takiego obejścia. W JS-ie często ludzie piszą `var self = this` i wtedy można normalnie odwoływać się do self. Natomiast w Typescripcie jak napiszesz funkcję `function(foo){this.bar  = 5}` to this zachowuje się jak w JSie, a jak napiszesz ze strzałką `(foo)=>{this.bar=5}` to this zachowuje się normalnie jak w javie. Można sobie obczaić tu: https://www.typescriptlang.org/play/